ZENKO-3301: Lifecycle Expiration

theocerutti opened the pull request:
- [ ] 1. Decouple rule type from Replication component - we might need to introduce a new “Workflow” component + tests.
(https://github.com/scality/zenko-ui/pull/165)

- [ ] 2. Create/Edit expiration workflow Redux logic + tests.

- [ ] 3. Expiration workflow “source” part logic + tests.

- [ ] 4. Expiration workflow creation/edition layout + tests.

- [ ] 5. List expiration workflow Redux logic + new row type + tests.

- [ ] 6. Delete expiration + confirmation message + tests.

theocerutti committed:
ZENKO-3568: Decouple rule type from Replication component: added Workflow component + tests

bert-e commented:
# Hello theocerutti,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

theocerutti committed:
ZENKO-3301: Add Creation/Edition/Deletion/Search redux logic for Expiration Rules

nicolas2bert commented on src/react/workflow/workflow/Workflow.jsx:
```diff
@@ -0,0 +1,132 @@
+//@flow
```
Let's move `src/react/workflow/workflow/Workflow.jsx` to `src/react/workflow/WorkflowEditor.jsx`

nicolas2bert commented on src/react/workflow/replication/Replication.jsx:
```diff
@@ -1,85 +1,53 @@
 // @flow
```
Let's move `src/react/workflow/replication/Replication.jsx` to `src/react/workflow/workflowTypes/Replication.jsx`.
We will have soon `src/react/workflow/workflowTypes/Expiration.jsx` for the expiration form.

nicolas2bert commented on src/react/workflow/details/Configuration.jsx:
```diff
@@ -62,7 +54,7 @@ function Configuration({ wfSelected, replications, bucketList, locations, showEd
                         </T.BannerContainer>
                         <Button icon={<i className="fas fa-trash" />} text="Delete Workflow" variant='buttonDelete' onClick={handleOpenDeleteModal} />
```
"Banner" and "deletion button" could be moved to `src/react/workflow/WorkflowEditor.jsx` and only be shown when createMode is set to `false`. That way we might avoid having to (re)define `showEditWorkflowNotification` in three deferent component, hence avoiding extra rendering.

elenamarcova replied:
Code review is expected from @nicolas2bert .

nicolas2bert reviewed (commented):
First iteration on commit: 9369d2da2ce6d6e12db1cb3cc038e659ad1594bd

theocerutti committed:
Move workflows to workflowTypes directory, move Banner/deletion button to WorkflowEditor.
Add Expiration Form Edition/Creation + Deletion.
Implements Expiration into Workflow Table List + Fix V2 component

bert-e commented:
# Incorrect Jira project



The Jira issue ZENKO-3301 specified in the source
branch name, does not belong to project `ZKUI`.
